kolibri/core/assets/src/api-resource.js

```diff
@@ -289,6 +300,15 @@ class Collection {
   }
 
   /**
+   * Clear this Collection's cache of models.
+   */
+  clearCache() {
+    // Reset current models.
+    this.models = [];
+    this._model_map = {};
```

Comment:
I've updated it to go with the conservative strategy, but to make malformed data behave the same as a fetch error (rejecting, rather than resolving), so I think that's consistent at least.